From d4dd7abff02c2628c50019ddfceffd28b0dc6e56 Mon Sep 17 00:00:00 2001
From: Andre Heinecke <aheinecke@intevation.de>
Date: Wed, 6 Apr 2011 15:07:25 +0000
Subject: [PATCH] Not only use hardcoded install path on Windows

    On Windows the Path setting from the shared-desktop *.ontology files
    is the install directory from the compile time.
    Which is not something like /usr/share/ontologies but something like
    r:\kde\build\shared-desktop-ontologies-mingw-i686\image\shared\ontologies
    so if you did not compile shared-desktop-ontologies yourself the Path
    is useless.
    If the file is not found in the specified Path location the
    ontologyLoader now looks for the .trig file in the same
    location where the .ontology files are placed which is the default.
---
 nepomuk/services/storage/ontologyloader.cpp |   18 ++++++++++++++++++
 1 files changed, 18 insertions(+), 0 deletions(-)

diff --git a/nepomuk/services/storage/ontologyloader.cpp b/nepomuk/services/storage/ontologyloader.cpp
index 729a031..fbb8cef 100644
--- a/nepomuk/services/storage/ontologyloader.cpp
+++ b/nepomuk/services/storage/ontologyloader.cpp
@@ -75,6 +75,24 @@ void Nepomuk::OntologyLoader::Private::updateOntology( const QString& filename )
     // only update if the modification date of the ontology file changed (not the desktop file).
     // ------------------------------------
     QFileInfo ontoFileInf( df.readEntry( QLatin1String("Path") ) );
+
+#ifdef Q_OS_WIN
+    if ( ! ontoFileInf.exists() ) {
+        // On Windows the Path setting is always the install directory which is not
+        // something like /usr/share/ontologies but something like
+        // r:\kderoot\build\shared-desktop-ontologies-mingw-i686\image\shared\ontologies
+        // so if you did not compile shared-desktop-ontologies yourself the Path
+        // is useless.
+        // We expect a default ontologies installation where the .ontology files are placed
+        // in the same directory as the .trig files.
+        QString alt_filename = filename;
+        QFileInfo ontoAlternative( alt_filename.replace( QLatin1String(".ontology"),
+                                                         QLatin1String(".trig") ) );
+        ontoFileInf = ontoAlternative;
+        kDebug() << "Ontology path: " << filename << " does not exist. Using "
+                 << alt_filename << " instead";
+    }
+#endif
     QString ontoNamespace = df.readEntry( QLatin1String("Namespace") );
     QDateTime ontoLastModified = model->ontoModificationDate( ontoNamespace );
     bool update = false;
-- 
1.7.0.4

